shell/platform/linux/fl_view.cc

```diff
@@ -153,6 +154,19 @@ static gboolean run_flutter_engine(FlView* self) {
   return TRUE;
 }
 
+static int64_t gdk_button_to_flutter_button(guint button) {
+  switch (button) {
+    case 1:
+      return kFlutterPointerButtonMousePrimary;
+    case 2:
+      return kFlutterPointerButtonMouseMiddle;
+    case 3:
+      return kFlutterPointerButtonMouseSecondary;
+    default:
+      return -1;
```

Comment:
Is the button number arbitrarily high in `GdkEventButton`? If so, instead of this function you can just send `1 << (button-1)` (per the comment at the end of the button definitions enum).

Reply:
Sadly this doesn't work as GTK/X numbers the buttons from left to right on a traditional three button mouse, i.e. primary, middle, secondary. Flutter does these as Primary, Secondary, Middle. And then back/forward and scrolls also probably get confusing...

Reply:
FWIW, I'm not sure if the "back" and "forward" labels are actually meaningful to Flutter (I suspect not). But I'm fine with just handling these for now and revisiting other buttons if it ever comes up.